packages/lesswrong/server/editor/make_editable_callbacks.ts

```diff
@@ -371,26 +371,23 @@ function addEditableCallbacks<T extends DbObject>({collection, options = {}}: {
       const userId = currentUser._id
       const editedAt = new Date()
       const changeMetrics = htmlToChangeMetrics("", html);
-      
-      // FIXME: This doesn't define documentId, because it's filled in in the
-      // after-create callback, passing through an intermediate state where it's
-      // undefined; and it's missing _id and schemaVersion, which leads to having
-      // ObjectID types in the database which can cause problems. Would be good
-      // to remove this ts-ignore, since there was recently an important bug here
-      // (missing fieldName) that typechecking would have caught.
-      // @ts-ignore
-      const firstRevision = await Connectors.create(Revisions, {
-        ...await buildRevision({
+      const newRevision: Omit<DbRevision, "documentId" | "schemaVersion" | "_id"> = {
+        ...(await buildRevision({
           originalContents: doc[fieldName].originalContents,
           currentUser,
-        }),
+        })),
         fieldName,
         collectionName,
         version,
-        updateType: 'initial',
+        updateType: "initial",
         commitMessage,
         changeMetrics,
-      });
+      };
+      // FIXME: This doesn't define documentId, because it's filled in in the
+      // after-create callback, passing through an intermediate state where it's
+      // undefined; and it's missing _id and schemaVersion, which leads to having
```

Comment:
Currently as phrased this is ambiguous between 'these are the bad effects that would happen if we didn't omit these fields' and 'we had to omit these fields (for some reason), which is leading to these bad effects'. After re-reading, I suspect the former, but maybe rephrase as something like `...it's also missing _id and schemaVersion, because if these fields are present it will create ObjectID types in the database...'`

Reply:
I'm not rightly sure, but it's Jim's comment unaltered, just moved, and I'm not inclined to clarify it without understanding more.